13/andreitch/directors.py

```diff
@@ -0,0 +1,71 @@
+import csv
+from collections import defaultdict, namedtuple
+from functools import reduce
+
+MOVIE_DATA = '../movie_metadata.csv'
+NUM_TOP_DIRECTORS = 20
+MIN_MOVIES = 4
+MIN_YEAR = 1960
+
+Movie = namedtuple('Movie', 'title year score')
+
+
+def get_movies_by_director():
+    '''Extracts all movies from csv and stores them in a dictionary
+    where keys are directors, and values is a list of movies (named tuples)'''
+    directors = defaultdict(list)
+    with open(MOVIE_DATA, newline='') as csvfile:
+        reader = csv.DictReader(csvfile)
+        for row in reader:
+            if row['title_year'] and int(row['title_year']) >= MIN_YEAR:
+                directors[row['director_name']].append(
+                    Movie(row['movie_title'], row['title_year'], float(row['imdb_score'])))
+    return directors
+
+
+def get_average_scores(directors):
+    '''Filter directors with < MIN_MOVIES and calculate averge score'''
+    new_dict = {}
+    for k, v in directors.copy().items():
+        if len(v) >= MIN_MOVIES:
+            new_dict[(k, _calc_mean(directors[k]))] = v
+    return new_dict
+
+
+def _calc_mean(movies):
+    '''Helper method to calculate mean of list of Movie namedtuples'''
+    return round(reduce(lambda a, b: a+b, [m.score for m in movies]) / len(movies), 1)
```

Comment:
it is ok to use functional programming and the reduce function, but I don't see the value here, I think it is easier to use `sum` and divide by the len? So `sum([m.score for m in movies])` should do the same?